src/prof_data.c

```diff
@@ -451,16 +451,13 @@ prof_thread_name_alloc(tsd_t *tsd, const char *thread_name) {
 	}
 
 	size = strlen(thread_name) + 1;
-	if (size == 1) {
-		return "";
-	}
-
 	ret = iallocztm(tsd_tsdn(tsd), size, sz_size2index(size), false, NULL,
```

Comment:
I would suggest either do iallocztm with parameter `zero = true` or explicitly make `ret [size - 1] = 0` just in case the name passed in does not end properly.

Reply:
Good point. Added the terminator below.